generators/server/prompts.js

```diff
@@ -369,6 +369,31 @@ function askForServerSideOpts(meta) {
             ],
             default: (applicationType === 'microservice' || applicationType === 'uaa') ? 1 : 0
         },
+        {
+            when: response => ((response.databaseType === 'no' || response.hibernateCache === 'no') && applicationType !== 'gateway'),
+            type: 'list',
+            name: 'cacheImplementation',
+            message: 'Do you want to use spring cache?',
+            choices: [
```

Comment:
Lets not try to add new cache providers now, first iteration should be to support ehcache, hazelcast and infinispan and later more can be added